### PR TITLE
[MM-14571] Add e2e to test @mentions notifications

### PR DESCRIPTION
**Pull request by thekiiingbob**
#### Summary
- Added e2e test to test at-mention notification
- Add plugin to cypress to use Mattermost Client4 for sending messages from other users
- Added id to unread mentions badge, updated snapshot
- Added axios package for use in Cypress plugin, updated package-lock.json

#### Ticket Link
[JIRA Ticket](https://mattermost.atlassian.net/browse/MM-14571)
Fixes https://github.com/mattermost/mattermost-server/issues/10441

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

**Comment by thekiiingbob** [comment]
Something I'm still playing with is the usage of the Client4. Ideally, I'd like to get an instance of it inside the tests themselves so we can do something like the following (which does not work):

```javascript
cy.task('getClient4', {user}).then(async (client4) => {
            await client4.createPost({
                channel_id: channelId,
                message,
                type: '',
            })
        })
```

or

```javascript
cy.task('getClient4', {user}).as('Client4');
        cy.get('@Client4').invoke('createPost', {
            channel_id: channelId,
            message,
            type: '',
        });
```

That said, what is in this PR works for now :)

File: cypress/plugins/postMessageAs.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    const token = response.headers.token;
+
+    Client4.setUrl(cypressConfig.baseUrl);
+    Client4.setToken(token);
+
+    return Client4.createPost({
+        channel_id: channelId,
+        message,
+        type: '',
+    });
```
Comment (saturninoabril) [inline cypress/plugins/postMessageAs.js]: I was thinking if this should be done in `cy.request` (wrapped in a new custom command) but then it'll loose the current session and will not be able to listen to Notification.  This turned out to be a better solution. 💯

File: package.json [hunk 2]
```diff
@@ -69,6 +69,7 @@
     "@babel/preset-env": "7.3.4",
     "@babel/preset-react": "7.0.0",
     "@babel/runtime": "7.3.4",
+    "axios": "^0.18.0",
```
Comment (saturninoabril) [inline package.json]: Please use exact version.
```suggestion
    "axios": "0.18.0",
```

**Comment by saturninoabril** [comment]
@thekiiingbob re: `Client4` that sounds a good plan.  However, I'm more incline to directly connect to API via `cy.request`. I'll submit some of those, probably next week.

**Comment by thekiiingbob** [comment]
@saturninoabril Done with the requested changes!

File: cypress/integration/at_mentions/at_mentions_spec.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            expect(title).to.equal('Town Square');
+
+            const body = `@${sender.username}: ${message}`;
+
+            // * Verify additional args of notification
+            expect(opts).to.include({body, tag: body, requireInteraction: false, silent: false});
+        });
+
+        // * Verify unread mentions badge
+        cy.get('#sidebarItem_town-square').
```
Comment (saturninoabril) [inline cypress/integration/at_mentions/at_mentions_spec.js]: Please add `scrollIntoView()`.  This sometimes gave me intermittent failing result.

Comment (thekiiingbob) [reply cypress/integration/at_mentions/at_mentions_spec.js]: @saturninoabril Fixed. We're good now. :)

**Pull request merged**